### ISSUE TITLE
Bigger resolution training

### ISSUE THREAD

**Issue by gracious-patience** [ByteDance-Seed/Bagel#127]
Hi! Thanks very much for your work!

I want to fine-tune Bagel to generate images with the max side > 1024. I changed this parameter for 1408 in `ImageTransform` and set `interpolate_pos` to True (because otherwise the training failed).

1. After I finish my training, will it be able to generate images with new side with `interpolate_pos` = False?
2. Should i also adjust the max side for vit_transform too?

**Comment by causalfusion** [comment]
Hi, you don't need to set `interpolate_pos=True` because we use extrapolate by default.  Instead, you should set `max_latent_size` to `1408 // 16`.

**Comment by gracious-patience** [comment]
#

Are you sure? I mean, i don't want to generate 1408х1408 images, but rather be able to generate images of different aspect ratios like 1408x768, which have approximately the same amount of pixels as 1024х1024 images.



> Hi, you don't need to set `interpolate_pos=True` because we use extrapolate by default. Instead, you should set `max_latent_size` to `1408 // 16`.

**Comment by causalfusion** [comment]
it should work. `max_latent_size` controls the maximum size, you are still able to generate images of different aspect ratios. You can have a try and see if there is any other problem.

**Comment by gracious-patience** [comment]
i have an error while loading pretrained checkpoint when setting `max_latent_size` to another value:
```
size mismatch for _fsdp_wrapped_module.bagel.latent_pos_embed._fsdp_wrapped_module.pos_embed: copying a param with shape torch.Size([4096, 3584]) from checkpoint, the shape in current model is torch.Size([7744, 3584]).
```
I guess i'll have to reinitialize `pos_embed`. Hope that won't corrupt the pretrained quality much and `pos_embed` will learn fast.

**Comment by gracious-patience** [comment]
> it should work. `max_latent_size` controls the maximum size, you are still able to generate images of different aspect ratios. You can have a try and see if there is any other problem.

It worked, thanks!

**Comment by fishing-pan** [comment]
# @gracious-patience 
Hello, when I try to infer at a higher resolution, I also encounter the error `Trying to set a tensor of shape torch.Size([4096, 3584]) in "pos_embed" (which has shape torch.Size([9216, 3584])), this looks incorrect.` Besides changing `max_latent_size`, what else should I change?

**Comment by fishing-pan** [comment]
> > it should work. `max_latent_size` controls the maximum size, you are still able to generate images of different aspect ratios. You can have a try and see if there is any other problem.
> 
> It worked, thanks!

Hello, when I try to infer at a higher resolution, I also encounter the error Trying to set a tensor of shape torch.Size([4096, 3584]) in "pos_embed" (which has shape torch.Size([9216, 3584])), this looks incorrect. Besides changing max_latent_size, what else should I change?
